src/hooks/useFastSearchFromOptions.ts

```diff
@@ -63,8 +72,22 @@ function useFastSearchFromOptions(
                 },
             },
         ]);
+        setFastSearch(newFastSearch);
+        previousFastSearchRef.current = newFastSearch;
+        return () => {
+            if (!previousFastSearchRef.current) {
+                return;
+            }
+            previousFastSearchRef.current.dispose();
+            previousFastSearchRef.current = null;
+        };
+    }, [options]);
 
+    const findInSearchTree = useMemo(() => {
```

Comment:
This should use `useCallback`
```suggestion
    const findInSearchTree = useCallback((searchInput: string): AllOrSelectiveOptions => {
```

Reply:
Good catch